lib/theory/src/Theory/Constraint/Solver/ProofMethod.hs

```diff
@@ -1041,7 +1079,7 @@ smartRanking :: ProofContext
              -> System
              -> [AnnotatedGoal] -> [AnnotatedGoal]
 smartRanking ctxt allowPremiseGLoopBreakers sys =
-    sortOnUsefulness . unmark . sortDecisionTree notSolveLast . sortDecisionTree solveFirst . goalNrRanking
+    moveNatToEnd . sortOnUsefulness . unmark . sortDecisionTree notSolveLast . sortDecisionTree solveFirst . goalNrRanking
```

Comment:
it puts it to the lowest priority. I think the reasoning was that unfolding it creates many case-distinctions which we usually don't need because of the special nat-based reasoning techniques